snuba/datasets/querylog_processor.py

```diff
@@ -0,0 +1,76 @@
+import simplejson as json
+from typing import Any, Mapping, Optional, Sequence, Union
+import uuid
+
+from snuba.processor import MessageProcessor, ProcessedMessage, ProcessorAction
+
+
+class QuerylogProcessor(MessageProcessor):
+    def __get_request(self, request: Mapping[str, Any]) -> str:
+        return json.dumps({k: v for k, v in sorted(request.items())})
+
+    def __get_sample(self, sample: Union[int, float]) -> float:
+        """
+        Since sample can be both integer or float between 0 and 1, we just cast
+        int to float so we have a common data type.
+        """
+        if isinstance(sample, int):
+            return float(sample)
+
+        return sample
+
+    def __extract_query_list(
+        self, query_list: Sequence[Mapping[str, Any]]
+    ) -> Mapping[str, Any]:
+        sql = []
+        status = []
+        final = []
+        cache_hit = []
+        sample = []
+        max_threads = []
+        duration_ms = []
+        trace_id = []
+
+        for query in query_list:
+            sql.append(query["sql"])
+            status.append(query["status"])
+            final.append(query["stats"].get("final", 0))
+            cache_hit.append(query["stats"].get("cache_hit", 0))
+            sample.append(query["stats"].get("sample", 0))
+            max_threads.append(query["stats"].get("max_threads", 0))
+            trace_id.append(str(uuid.UUID(query["trace_id"])))
```

Comment:
is this because the trace_id does not come in the right format ?

Reply:
Yes, it doesn't have hyphens